### PR TITLE
[ARC-1486] Adding e2e tests

### PR DESCRIPTION
**Pull request by mboudreau**
**What's in this PR?**
E2E Test framework working

**Why**
Because pollinator ain't cutting it

**Added feature flags**
None

**Affected issues**  
ARC-1486

**How has this been tested?**  
I mean, I added tests

**Whats Next?**
After this is working in main, need to add more tests to replace pollinator

### CODE REVIEW

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -23,35 +20,107 @@ jobs:
...
       - run: yarn run build:release
 
-  test:
-    runs-on: ubuntu-latest
+  unit-test:
+    runs-on: ubuntu-22.04
 
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-node@v3
```
Comment (gxueatlassian) [inline .github/workflows/on-push.yml:31]: 👍

File: .github/workflows/on-push.yml [hunk 2]
```diff
@@ -23,35 +20,107 @@ jobs:
...
+      - name: create .env file
+        run: |
+          echo "${{ secrets.E2E_GITHUB_PRIVATE_KEY }}" > jira-test.pem
+          echo "APP_URL=http://localhost" > .env
+          echo "WEBHOOK_PROXY_URL=http://localhost" >> .env
+          echo "INSTANCE_NAME=fusion-test" >> .env
+          echo "NODE_ENV=test" >> .env
+          echo "APP_NAME=jira" >> .env
+          echo "APP_ID=${{ secrets.E2E_GITHUB_APP_ID }}" >> .env
+          echo "WEBHOOK_SECRET=${{ secrets.E2E_GITHUB_WEBHOOK_SECRET }}" >> .env
```
Comment (gxueatlassian) [inline .github/workflows/on-push.yml:43]: The step is called `unit-test`, but the secret is called `E2E_xxxx`, seems a bit strange to me. Am I missing something?

File: src/config/env.ts [hunk 3]
```diff
@@ -63,11 +43,26 @@ export const envVars: EnvVars = new Proxy<object>({}, {
...
 
-// Check to see if all required environment variables are set
-const missingVars = requiredEnvVars.filter(key => envVars[key] === undefined);
-if (missingVars.length) {
-	throw new Error(`Missing required Environment Variables: ${missingVars.join(", ")}`);
-}
+envCheck(
+	"APP_ID",
+	"APP_URL",
+	"INSTANCE_NAME",
```
Comment (gxueatlassian) [inline src/config/env.ts:49]: `INSTANCE_NAME` is new in the check compare to existing code. Just to confirm this is intended correct?

**Review by gxueatlassian — APPROVED** [review]
I am happy with the code structure and the tests so far.
Happy to get involved in more discussion though if anyone feels like to discuss this PR.

File: .github/workflows/on-push.yml (hunk 2 above)
Comment (mboudreau) [reply .github/workflows/on-push.yml:43]: nah, just copy pasted it from the e2e tests.  Not sure this is even needed as unit tests never go out to the internet.  I'll review this

File: src/config/env.ts (hunk 3 above)
Comment (mboudreau) [reply src/config/env.ts:49]: Yes, it's needed or else stuff fails, was just something that was overlooked in the past.

**Comment by gxueatlassian** [comment]
Hi @mboudreau just have a quick one, once merged, will each PR's build conflicting each other (since they are installing/uninstalling same app? )

File: test/e2e/utils/github.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { Page } from "@playwright/test";
+import { GithubTestDataRoles, testData } from "test/e2e/constants";
+import { e2eEnvVars } from "test/e2e/env-e2e";
+
+const data = testData.github;
+export const githubLogin = async (page: Page, roleName: keyof GithubTestDataRoles, saveState = false) => {
```
Comment (joshkay10) [inline test/e2e/utils/github.ts:6]: This looks like it is never used? it's commented out in setup.ts, is that just commented while you solve it or didn't work?

File: test/e2e/utils/jira.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	const role = data.roles[roleName];
+	if (!role.username || !role.password) {
+		throw "Jira username or password missing";
+	}
+	await page.goto(data.urls.login);
+	await page.waitForLoadState();
+	// This is a hack because atlassian auth is all frontend, it shows the login page and form
+	// but then redirects to the correct page if the token is available.  However, playwright
+	// will continue after the page loads and try to fill in the form before being redirected,
+	// causing a flow issue.  This is non-standard for any authentication as it should
```
Comment (joshkay10) [inline test/e2e/utils/jira.ts:17]: 🤣

File: test/setup/env-test.ts [hunk 6]
```diff
@@ -1,5 +1,11 @@
-import { EnvVars } from "config/env";
+import {  EnvVars } from "config/env";
```
Comment (joshkay10) [inline test/setup/env-test.ts:1]: /nit/nit Double space 🌈 🌈

**Review by joshkay10 — APPROVED** [review]
Gee wizz, a lot of stuff going on there. Would definitely be keen for a walkthrough too

File: test/e2e/utils/github.ts (hunk 4 above)
Comment (mboudreau) [reply test/e2e/utils/github.ts:6]: it works, but it gets blocked and asks for 2FA.  The next PR will have that in, this is just the start :)

**Comment by mboudreau** [comment]
@gxueatlassian Yes, the e2e tests can collide.  This is just the base and the next bit is to start building a new app for every run so they don't collide.  right now though, the tests are just installing the app, so there's no much there in terms of functionality tested, I just wanted to get this in first so the scope of this PR doesn't blow out

**Pull request merged**